### PR TITLE
Feature/4/stretch

### PR DESCRIPTION
**Pull request by monsieurtanuki**
It's a mess with github - sorry about that.
The only interesting commit is the ["Stretched text"](https://github.com/ookami-kb/flutter_arc_text/commit/9dd89612512855c1a2ac39d5a6b89598c9c64b89) one.

### CODE REVIEW

File: lib/src/arc_text_painter.dart [hunk 1]
```diff
@@ -13,40 +13,52 @@ class ArcTextPainter {
...
+    double stretchAngle,
+    double interLetterAngle,
   })  : assert(radius != null, 'radius should not be null'),
         assert(text != null, 'text should not be null'),
         assert(textStyle != null, 'textStyle should not be null'),
         assert(alignment != null, 'alignment should not be null'),
         assert(initialAngle != null, 'initialAngle should not be null'),
         assert(direction != null, 'direction should not be null'),
         assert(placement != null, 'placement should not be null'),
+        assert(!(stretchAngle != null && interLetterAngle != null),
```
Comment (ookami-kb) [inline lib/src/arc_text_painter.dart:25]: I would say the inversed condition would be slightly more readable.

```suggestion
        assert(stretchAngle == null || interLetterAngle == null,
```

File: lib/src/arc_text_painter.dart [hunk 2]
```diff
@@ -13,40 +13,52 @@ class ArcTextPainter {
...
+        _angleWithAlignment = initialAngle - alignmentOffset + math.pi;
         _angleMultiplier = -1;
-        _heightOffset = this._radius - _textPainter.height;
+        _heightOffset = this._radius;
         break;
     }
+
+    if (stretchAngle != null && _text.runes.length > 1) {
+      _interLetterAngle =
+          (stretchAngle - finalAngle) / (_text.runes.length - 1);
```
Comment (ookami-kb) [inline lib/src/arc_text_painter.dart:60]:
```suggestion
      _interLetterAngle = (stretchAngle - finalAngle) / _text.runes.length;
```

I think without `-1` it looks better, at least with angle `360` it seems more natural:

image

**Comment by monsieurtanuki** [comment]
@ookami-kb As I said before, I got trapped in a github black hole.
I have no idea which files or which changes you're referring too.
If needed, feel free to just copy the interesting part of my code, put that in a PR created by you and merge it: I won't be offended.

**Comment by ookami-kb** [comment]
Cool, copied to #9

**Comment by monsieurtanuki** [comment]
@ookami-kb Well done!
One suggestion: I would have added something like a slider for the stretch angle in the `pub.dev` screenshot, with a value like 75% (`*2*math.pi` of course).
Or just added the screenshot of a short `2*math.pi` text like "I feel stretched".

**Comment by ookami-kb** [comment]
Yeah, it makes sense. Currently, there's a link to the interactive demo in the readme file: https://ookami-kb.github.io/flutter_arc_text/ where you can play with all the parameters, but I'll add a screenshot as well.

**Comment by monsieurtanuki** [comment]
Nice demo. I would add 2 checkboxes: "stretch" and "display circle".

**Comment by ookami-kb** [comment]
`stretch = 0` basically corresponds to "no stretch" as there's no practical sense in having stretching to 0 deg.

"Display circle" – yeah, that can be useful.

**Comment by monsieurtanuki** [comment]
stretch = 0 or 1 doesn't look nice at all, we agree, but it's not like stretch = null where the normal letter spacing takes place.

**Comment by ookami-kb** [comment]
In demo `stretch = 0` is replaced with `stretch = null`.

**Comment by monsieurtanuki** [comment]
Oh, I didn't notice that, I thought I went beyond the slider range that came back to its initial value.
Regardless, it's not obvious and it's supposed to be a demo.
I would make it obvious: stretch = 0 means something (could be used in an animation?), and stretch = null means something else.
As stretch = null was the previous default I assumed it was an option worth showing, but perhaps using a valid value for stretch is more meaningful now. I don't know.

**Pull request closed without merging**